src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp

```diff
@@ -219,6 +219,23 @@ TEST_P(OVInferRequestIOTensorTest, InferStaticNetworkSetChangedOutputTensorThrow
     ASSERT_ANY_THROW(req.infer());
 }
 
+TEST_P(OVInferRequestIOTensorTest, CheckInferIsNotChangeInput) {
+    ov::Tensor input_tensor;
+    OV_ASSERT_NO_THROW(input_tensor = req.get_tensor(input));
+
+    OV_ASSERT_NO_THROW(req.infer());
+
+    ov::Tensor input_after_infer;
+    OV_ASSERT_NO_THROW(input_after_infer = req.get_tensor(input));
+    ASSERT_EQ(input_tensor.data(), input_after_infer.data());
+
+    OV_ASSERT_NO_THROW(req.infer());
+
+    ov::Tensor input_after_several_infer;
+    OV_ASSERT_NO_THROW(input_after_several_infer = req.get_tensor(input));
+    ASSERT_EQ(input_tensor.data(), input_after_several_infer.data());
```

Comment:
can we test the data itself?
Here we check that pointer is not reallocated, but we need to check that values are the same as well

Reply:
yes, comparison of values is added

Reply:
I meant check that `data()` contains the same values

Reply:
moved to comparison with ov::test::utils::compare , it is checked values in this function